core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -112,6 +111,16 @@
         Setting.byteSizeSetting("index.translog.flush_threshold_size", new ByteSizeValue(512, ByteSizeUnit.MB), Property.Dynamic,
             Property.IndexScope);
 
+    /**
+     * The maximum size of a translog generation. This is independent of the maximum size of
+     * translog operations that have not been flushed.
+     */
+    public static final Setting<ByteSizeValue> INDEX_TRANSLOG_GENERATION_THRESHOLD_SIZE_SETTING =
+            Setting.byteSizeSetting(
+                    "index.translog.generation_threshold_size",
```

Comment:
I wanna understand why this is `64MB` why can't we use `INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING` or it's default instead rather than an arbitrary value? I would love to get some insight that we also can document? Then there is also the question why we can't simply flush this from the outside via `IndexShard#maybeFlush`, that would simplify the translog potentially, one property that I liked about the `Translog#add()` operation was that it would never acquire a write lock. We are now entering potentially dangerous territory here, locking can be a beast.